### ISSUE TITLE
Web App memory leak

### ISSUE THREAD

**Issue by rblaine95** [didx-xyz/aries-cloudapi-python#528]
There appears to be a severe memory leak in the Web App.

Relates to:
- #526 
- https://github.com/pydantic/pydantic/discussions/7912

**Comment by rblaine95** [comment]
In #526 @wdbasson set up `tracemalloc` in the `/admin/tenants` function to create tenants.

Here are the log outputs with Datadog:
```
# Create 100 holders

## Holder 1
/usr/local/lib/python3.9/pathlib.py:79: size=2560 KiB, count=1, average=2560 KiB
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:205: size=998 KiB, count=7597, average=135 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/profiling/collector/__init__.py:42: size=538 KiB, count=10504, average=52 B
<string>:1: size=493 KiB, count=6317, average=80 B
/usr/local/lib/python3.9/pathlib.py:728: size=482 KiB, count=3699, average=133 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/profiling/collector/memalloc.py:164: size=303 KiB, count=5017, average=62 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:580: size=266 KiB, count=2970, average=92 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:348: size=199 KiB, count=1532, average=133 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/profiling/exporter/http.py:175: size=173 KiB, count=2627, average=67 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/internal/packages.py:111: size=144 KiB, count=2, average=72.2 KiB
INFO:     10.1.105.20:37176 - "POST /admin/tenants HTTP/1.1" 200 OK

## Holder 100
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:205: size=5815 KiB, count=46339, average=128 B
/usr/local/lib/python3.9/pathlib.py:79: size=2560 KiB, count=1, average=2560 KiB
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:348: size=1271 KiB, count=10403, average=125 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/profiling/collector/__init__.py:42: size=666 KiB, count=12920, average=53 B
<string>:1: size=639 KiB, count=8174, average=80 B
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/ddtrace/profiling/collector/memalloc.py:164: size=608 KiB, count=10491, average=59 B
/usr/local/lib/python3.9/pathlib.py:728: size=482 KiB, count=3699, average=133 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:962: size=451 KiB, count=4754, average=97 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:673: size=447 KiB, count=6227, average=73 B
/usr/local/lib/python3.9/site-packages/aiohttp/client_proto.py:213: size=286 KiB, count=1962, average=149 B
INFO:     10.1.105.20:41784 - "POST /admin/tenants HTTP/1.1" 200 OK
```

Without Datadog:
```
## Holder 101
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:205: size=329 KiB, count=2497, average=135 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:348: size=71.0 KiB, count=555, average=131 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:962: size=28.1 KiB, count=334, average=86 B
/usr/local/lib/python3.9/abc.py:123: size=22.0 KiB, count=254, average=89 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:580: size=21.0 KiB, count=292, average=74 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:673: size=20.6 KiB, count=304, average=69 B
/usr/local/lib/python3.9/site-packages/pydantic/plugin/_schema_validator.py:34: size=11.6 KiB, count=21, average=568 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:1632: size=11.5 KiB, count=70, average=168 B
/usr/local/lib/python3.9/inspect.py:2266: size=10.0 KiB, count=61, average=168 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_model_construction.py:607: size=10120 B, count=115, average=88 B
INFO:     10.1.105.20:42878 - "POST /admin/tenants HTTP/1.1" 200 OK

## Holder 149
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:205: size=2620 KiB, count=20900, average=128 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:348: size=581 KiB, count=4771, average=125 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:962: size=209 KiB, count=2227, average=96 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:673: size=197 KiB, count=2771, average=73 B
/usr/local/lib/python3.9/site-packages/aiohttp/client_proto.py:213: size=129 KiB, count=946, average=140 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:1632: size=98.1 KiB, count=598, average=168 B
/usr/local/lib/python3.9/inspect.py:2266: size=84.4 KiB, count=512, average=169 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:870: size=79.2 KiB, count=2028, average=40 B
/usr/local/lib/python3.9/tracemalloc.py:67: size=74.6 KiB, count=1194, average=64 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_metadata.py:87: size=69.9 KiB, count=1119, average=64 B
INFO:     10.1.124.32:43102 - "POST /admin/tenants HTTP/1.1" 200 OK

## Holder 150
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:205: size=2698 KiB, count=21526, average=128 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_utils.py:348: size=596 KiB, count=4892, average=125 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:962: size=213 KiB, count=2271, average=96 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:673: size=202 KiB, count=2848, average=73 B
/usr/local/lib/python3.9/site-packages/aiohttp/client_proto.py:213: size=134 KiB, count=984, average=139 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:1632: size=99.9 KiB, count=609, average=168 B
/usr/local/lib/python3.9/inspect.py:2266: size=87.2 KiB, count=529, average=169 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py:870: size=80.8 KiB, count=2068, average=40 B
/usr/local/lib/python3.9/site-packages/pydantic/_internal/_core_metadata.py:87: size=71.3 KiB, count=1141, average=64 B
/usr/local/lib/python3.9/tracemalloc.py:67: size=69.6 KiB, count=1114, average=64 B
INFO:     10.1.105.20:50278 - "POST /admin/tenants HTTP/1.1" 200 OK
```

**Comment by ff137** [comment]
BTW, I tried with the help of ChatGPT to create a script to isolate pydantic memory usage ... and results seem to indicate that Pydantic is not the problem. I suspect it is the cloudcontroller interface that is somehow holding onto memory.

Here's the script I tested:
```py
from pydantic import BaseModel, Field
from typing import List, Optional
from uuid import uuid4
import tracemalloc


# Your custom model
class CreateTenantRequest(BaseModel):
    wallet_label: str
    wallet_name: Optional[str]
    roles: Optional[List[str]] = None
    group_id: Optional[str] = None
    image_url: Optional[str] = None


# cloudcontroller's model
class CreateWalletRequest(BaseModel):
    extra_settings: Optional[str] = None
    image_url: Optional[str] = None
    key_management_mode: Optional[str] = None
    label: Optional[str] = None
    wallet_dispatch_type: Optional[str] = None
    wallet_key: Optional[str] = None
    wallet_key_derivation: Optional[str] = None
    wallet_name: Optional[str] = None
    wallet_type: Optional[str] = None
    wallet_webhook_urls: Optional[List[str]] = None


class CreateWalletRequestWithGroups(CreateWalletRequest):
    group_id: Optional[str]


def simulate_cloudcontroller_api_call(data):
    # Simulating an API call that would create a wallet and return a response
    wallet_response = CreateWalletRequestWithGroups(**data)
    return wallet_response


def create_tenant(data):
    tenant_request = CreateTenantRequest(**data)
    # Simulating some operations similar to your FastAPI endpoint
    wallet_data = {
        "image_url": tenant_request.image_url,
        "key_management_mode": "managed",
        "label": tenant_request.wallet_label,
        "wallet_key": "some_random_key",
        "wallet_name": tenant_request.wallet_name or uuid4().hex,
        "wallet_type": "askar",
        "group_id": tenant_request.group_id,
    }
    wallet_response = simulate_cloudcontroller_api_call(wallet_data)
    return wallet_response


def main():
    tracemalloc.start()

    # Example data that would be sent to your endpoint
    data = {
        "wallet_label": "wallet1",
        "wallet_name": "wallet123",
        "roles": ["issuer"],
        "group_id": "group1",
        "image_url": "http://example.com/image.png",
    }

    # Loop to simulate repeated calls to your endpoint
    for _ in range(10000):
        create_tenant(data)

    print("Memory usage after 1000 tenants created:")
    snapshot = tracemalloc.take_snapshot()
    top_stats = snapshot.statistics("lineno")
    for stat in top_stats[:10]:
        print(stat)


if __name__ == "__main__":
    main()
```

Output:
```
Memory usage after 10000 tenants created:
../isolate_pydantic_script_cc.py:40: size=127 B, count=2, average=64 B
../isolate_pydantic_script_cc.py:34: size=41 B, count=2, average=20 B
../isolate_pydantic_script_cc.py:69: size=28 B, count=1, average=28 B
../isolate_pydantic_script_cc.py:63: size=8 B, count=1, average=8 B
```

**Comment by rblaine95** [comment]
Resolved via #548
